Classes/Backend/Form/ProfileShowFieldsItemProcFunc.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Fgtclb\AcademicPersons\Backend\Form;
+
+use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\Information\Typo3Version;
+
+class ProfileShowFieldsItemProcFunc
+{
+    /**
+     * @param array<string, mixed> $params
+     *
+     * @return void
+     */
+    public function showFields(&$params): void
+    {
+        // ToDo: Create a registry option to define more fields to select
+        $params['items'] = array_merge($this->defaultFieldItemShowFields(), []);
+        $params['itemGroups'] = [
+            'contracts' => 'Contracts',
+        ];
+    }
+
+    /**
+     * @return array<int, array<string|int, mixed>>
+     */
+    private function defaultFieldItemShowFields(): array
+    {
+        return [
+            [
+                ((new Typo3Version())->getMajorVersion() >= 12 ? 'label' : 0) => BackendUtility::getItemLabel('tx_academicpersons_domain_model_contract', 'position'),
```

Comment:
I'd suggest to build all 4 keys once in a variable and use only variables
to create the array. Makes it a minimal more readable.

On the other hand `main` is v11 only until now - but in general fine to make
that adjustable for now.

Another way could be to have either a small helper method, anon or provite to
call with 4 arguments which returns one entry:

```php
return [
  $this->createItem(
    BackendUtility::getItemLabel('tx_academicpersons_domain_model_contract', 'organisational_unit'),
    'contracts.organisationalUnit',
    null,
    'contracts',
  ),
  $this->createItem(...),
  $this->createItem(...),
  $this->createItem(...),
  // ...
];
```

and handle the version assoc/indexed array item balues within the method.

Reply:
And no fallback handling - can eventually lead to empty labels for the BU::getItemLabel() calls. Suggest sane `?:` fallbacks on that, just to be sure.